yb-voyager/src/tgtdb/attr_name_registry.go

```diff
@@ -12,45 +12,60 @@ import (
 )
 
 type AttributeNameRegistry struct {
-	dbType    string
-	tdb       TargetDB
-	tconf     *TargetConf
-	attrNames *utils.StructMap[sqlname.NameTuple, []string]
-	mu        sync.Mutex
+	dbType                  string
+	tdb                     TargetDB
+	tconf                   *TargetConf
+	attrNames               *utils.StructMap[sqlname.NameTuple, []string]
+	bestMatchingColumnCache *utils.StructMap[sqlname.NameTuple, map[string]string]
+	mu                      sync.Mutex
 }
 
 func NewAttributeNameRegistry(tdb TargetDB, tconf *TargetConf) *AttributeNameRegistry {
 	return &AttributeNameRegistry{
-		dbType:    tconf.TargetDBType,
-		tdb:       tdb,
-		tconf:     tconf,
-		attrNames: utils.NewStructMap[sqlname.NameTuple, []string](),
+		dbType:                  tconf.TargetDBType,
+		tdb:                     tdb,
+		tconf:                   tconf,
+		attrNames:               utils.NewStructMap[sqlname.NameTuple, []string](),
+		bestMatchingColumnCache: utils.NewStructMap[sqlname.NameTuple, map[string]string](),
 	}
 }
 
 func (reg *AttributeNameRegistry) QuoteAttributeName(tableNameTup sqlname.NameTuple, columnName string) (string, error) {
 	var err error
+	reg.mu.Lock()
```

Comment:
This function would be called by every goroutine for every event, so ideally, we should not lock here. Locking inside the `!ok` block is okay because that will run only once per table. (first event)